src/coreclr/src/debug/debug-pal/unix/diagnosticsipc.cpp

```diff
@@ -53,14 +53,28 @@ IpcStream::DiagnosticsIpc *IpcStream::DiagnosticsIpc::Create(const char *const p
     sockaddr_un serverAddress{};
     serverAddress.sun_family = AF_UNIX;
 
-    const ProcessDescriptor pd = ProcessDescriptor::FromCurrentProcess();
-    PAL_GetTransportName(
-        sizeof(serverAddress.sun_path),
-        serverAddress.sun_path,
-        pIpcName,
-        pd.m_Pid,
-        pd.m_ApplicationGroupId,
-        "socket");
+    // A user specified socket location and name supersedes the default
+    // we are in charge of freeing this memory
+    const char *customTransportPath = getenv("COMPlus_DiagnosticsServerTransportPath");
+    if (customTransportPath != nullptr)
+    {
+        int chars = snprintf(serverAddress.sun_path, sizeof(serverAddress.sun_path), "%s", customTransportPath);
+        _ASSERTE(chars > 0 && (unsigned int)chars < sizeof(serverAddress.sun_path));
+        free(customTransportPath);
+    }
+    else
+    {
+        // generate the default socket name in TMP Path
+        const ProcessDescriptor pd = ProcessDescriptor::FromCurrentProcess();
```

Comment:
It isn't ideal that it will be an either/or choice between connecting via PID or connecting via a custom path. I think it is a reasonable place to start though and we can wait for feedback. The ideal (which would take a good bit more work) is to listen to both the custom and default pipes simultaneously rather forcing a choice between them.

Reply:
I agree.  I'll look into what it would take to do this.  I imagine we could just create both sockets and then use a `select` like construct to call `accept` on both.

Reply:
> I'll look into what it would take to do this

Don't spend too much time on it : ) I just mentioned it so everyone was aware, not a request that we implement it right now.